TezosKit/Client/TezosNodeClient.swift

```diff
@@ -538,70 +537,73 @@ public class TezosNodeClient: AbstractClient {
    * This method parallelizes fetches to get chain and address data and returns all required data
    * together as an OperationData object.
    */
-  private func getMetadataForOperation(address: String) -> OperationMetadata? {
-    let fetchersGroup = DispatchGroup()
-
-    // Fetch data about the chain being operated on.
-    var chainID: String?
-    var headHash: String?
-    var protocolHash: String?
-    let chainHeadRequestRPC = GetChainHeadRPC { json, _ in
-      if let json = json,
-        let fetchedChainID = json["chain_id"] as? String,
-        let fetchedHeadHash = json["hash"] as? String,
-        let fetchedProtocolHash = json["protocol"] as? String {
-        chainID = fetchedChainID
-        headHash = fetchedHeadHash
-        protocolHash = fetchedProtocolHash
+  private func getMetadataForOperation(address: String) -> Promise<OperationMetadata> {
+    return Promise { seal in
+      let fetchersGroup = DispatchGroup()
```

Comment:
don't use dispatch groups anymore when using promises, you should update GetChainHeadRPC to return a promise and use a 
when(fulfilled: ) 
to combine all the promises

Reply:
Done. I can't update this code if I'm keeping promises outside of the core library, so left as is. 

I do realize what you are talking about though, I was going to try to stage promisekit integration out over a few CLs in order to make it easier to review.